WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerAdapter.java

```diff
@@ -541,16 +541,22 @@ public int compare(SiteRecord site1, SiteRecord site2) {
             }
 
             if (mSites == null || !mSites.isSameList(sites)) {
-                mAllSites = (SiteList) sites.clone();
-                mSites = filteredSitesByTextIfInSearchMode(sites);
+                SiteList allSites = (SiteList) sites.clone();
+                SiteList filteredSites = filteredSitesByTextIfInSearchMode(sites);
+
+                return new SiteList[]{allSites, filteredSites};
             }
 
             return null;
         }
 
         @Override
-        protected void onPostExecute(Void results) {
-            notifyDataSetChanged();
+        protected void onPostExecute(SiteList[] updatedSiteLists) {
+            if (updatedSiteLists != null) {
```

Comment:
`onPostExecute()` is already executing in main thread (by design) and `notifyDatSetChanged()` will execute in the same thread as well. So, I'd say there is no need for guarding these in a `synchronized` block and will only be "noise" in this case. Would you mind if we leave it as is? Thanks!